### PR TITLE
Loyal adblocking survey

### PR DESCRIPTION
**Pull request by Calanthe**
## What does this change?

The new ongoing adblocking survey (for loyal users only). The overlay with a CTA text and a survey link will be disable. Once disabled, it won't be shown again.
## What is the value of this and can you measure success?

We want to understand what causes people to block ads on theguardian.com and what would make them consider unblocking.
## Does this affect other platforms - Amp, Apps, etc?

no
## Screenshots

screen shot 2016-03-31 at 15 40 35
## 

_Does this PR meet the [contributing guidelines](https://github.com/guardian/frontend/blob/issue_pr_templates/.github/CONTRIBUTING.md#submission)?_

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+                marque36icon: svgs('marque36icon'),
+                crossIcon: svgs('crossIcon'),
+                surveyOverlaySimple: svgs('surveyOverlaySimple')
+            });
+    };
+
+    surveySimple.prototype.attach = function () {
+        if (!this.hasSeen()) {
+            fastdom.write(function () {
+                $(document.body).append(this.bannerTmpl);
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js]: Shouldn't this be `.append(this.bannerTmpl())` ?

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+            }.bind(this));
+        }
+    };
+
+    surveySimple.prototype.hasSeen = function () {
+        var messageStates = userPrefs.get(this.prefs);
+        return messageStates && messageStates.indexOf(this.id) > -1;
+    };
+
+    surveySimple.prototype.remember = function () {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js]: I think this method name is maybe a little broad? I personally would call the method `closePermanently` and the config item something else (like `shouldClosePermanently`, maybe)

File: static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+        return adblockUsed ? 'https://surveys.theguardian.com/R.aspx?a=684&as=AC5tF8XE6G&t=1' + ophanId : 'https://surveys.theguardian.com/R.aspx?a=685&as=Aj5jo96J9j&t=1' + ophanId;
+    }
+
+    return function () {
+        this.id = 'LoyalAdblockingSurvey';
+        this.start = '2016-03-21';
+        this.expiry = '2016-05-31';
+        this.author = 'Zofia Korcz';
+        this.description = 'An adblock ongoing survey for all loyal users';
+        this.audience = 0;
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js]: There's no audience?

File: static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+        this.canRun = function () {
+            return true;
+        };
+
+        this.variants = [{
+            id: 'variant',
+            test: function () {
+                detect.adblockInUse.then(function (adblockUsed) {
+                    var alreadyVisited = storage.local.get('gu.alreadyVisited') || 0;
+                    if (alreadyVisited > 10) {
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js]: Shouldn't this be `alreadyVisited > 10 && adblockUsed`? The promise will resolve with a boolean whether or not an adblocker is found.

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js (hunk 1 above)
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js]: No, why? `this.bannerTmpl` is not a function. It's a result of what `template` function returns.

File: static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+    $,
+    config,
+    detect,
+    storage,
+    userFeatures,
+    SurveySimple
+) {
+    function getSurveyLink(adblockUsed) {
+        var ophanId = '&ophanId=' + config.ophan.pageViewId;
+        return adblockUsed ? 'https://surveys.theguardian.com/R.aspx?a=684&as=AC5tF8XE6G&t=1' + ophanId : 'https://surveys.theguardian.com/R.aspx?a=685&as=Aj5jo96J9j&t=1' + ophanId;
```
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js]: So we have a survey for both people who block and people who do not?

File: static/src/stylesheets/module/commercial/surveys/_survey.scss [hunk 6]
```diff
@@ -12,3 +12,300 @@
...
+            margin-top: $gs-gutter;
+
+            @include mq(tablet) {
+                margin-top: 0;
+            }
+        }
+
+    }
+
+    .subscriber-number-form {
```
Comment (jbreckmckye) [inline static/src/stylesheets/module/commercial/surveys/_survey.scss]: Do we need to restore this class?

File: static/src/stylesheets/module/commercial/surveys/_survey.scss [hunk 7]
```diff
@@ -12,3 +12,300 @@
...
+    }
+
+    .inline-arrow-white-right {
+        display: block;
+        float: right;
+        height: $gs-column-width/2;
+    }
+}
+
+.live-events-survey-wrapper {
```
Comment (jbreckmckye) [inline static/src/stylesheets/module/commercial/surveys/_survey.scss]: Is this class necessary?

File: static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js (hunk 1 above)
Comment (jbreckmckye) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js]: Doesn't _.template just return another function?

Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/commercial/survey/survey-simple.js]: In version 2.4.2, it returns a string.

File: static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js (hunk 3 above)
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js]: Not yet. First I wanted to set up an AB test with no audience, so we can test it by opting in manually.

File: static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js (hunk 5 above)
Comment (Calanthe) [inline static/src/javascripts/projects/common/modules/experiments/tests/loyal-adblocking-survey.js]: Yes, there are 2 different surveys.

**Comment by jbreckmckye** [comment]
:+1:

**Pull request merged**